plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -505,86 +499,112 @@ public void pull(final PullParams params, final ProgressMonitor progressMonitor)
-                                                           .withNewMetadata()
-                                                               .withName(newImageStreamName)
-                                                           .endMetadata()
-                                                           .withNewTag()
-                                                               .withNewFrom()
-                                                                   .withKind("DockerImage")
-                                                                   .withName(sourceImageWithTag)
-                                                               .endFrom()
-                                                           .endTag()
-                                                           .done();
+            String sourceImageTagName = KubernetesStringUtils.getTagNameFromRepoString(sourceImage);
+            ImageStreamTag existingTag = getImageStreamTagFromRepo(sourceImageTagName);
+            sourceImageWithTag = existingTag.getTag().getFrom().getName();
+        } catch (IOException e) {
+            // Image not found.
+            sourceImageWithTag = String.format("%s:%s", sourceImage, tag);
+        }
 
-            // Wait up to 5 seconds for image metadata to be pulled
-            for (int waitCount = 0; waitCount < OPENSHIFT_IMAGESTREAM_MAX_WAIT; waitCount++) {
-                Thread.sleep(1000);
+        String imageStreamTagName = KubernetesStringUtils.getImageStreamTagName(sourceImageWithTag,
+                                                                                targetImage);
 
-                ImageStreamTag createdTag = openShiftClient.imageStreamTags()
-                                                           .inNamespace(openShiftCheProjectName)
-                                                           .withName(newImageStreamName)
-                                                           .get();
-                if (createdTag != null) {
-                    LOG.info(String.format("Created ImageStreamTag %s in namespace %s",
-                                           imageStreamTag.getMetadata().getName(),
-                                           openShiftCheProjectName));
-                    return;
-                }
-            }
+        createImageStreamTag(sourceImageWithTag, imageStreamTagName);
+    }
 
-            throw new ImageNotFoundException(String.format("Image %s not found.", sourceImageWithTag));
 
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new IOException(e.getLocalizedMessage(), e);
-        }
-    }
 
     @Override
     public ImageInfo inspectImage(InspectImageParams params) throws IOException {
 
         String image = params.getImage();
+        if (image.contains(":")) { // Image to inspect passed in with tag, which we don't support
```

Comment:
This code is duplicated in removeImage too. You may want to extract it to a new method